### ISSUE TITLE
[Bug]: Failing Transactions don't display a Warning anymore and set an insane amount of Gas fees

### ISSUE THREAD

**Issue by seaona** [MetaMask/metamask-extension#17379]
### Describe the bug

**Problem**: when a tx is going to fail, it displays an insane amount of Gas fees and no warning:

![image](https://user-images.githubusercontent.com/54408225/214341027-cbe66431-6571-4cdc-a83d-68ef2a4628f2.png)

We used to display a warning, like this:

![image](https://user-images.githubusercontent.com/54408225/214340221-5e801abf-0cd0-4702-98cc-e9d310cc9af3.png)

### Steps to reproduce

1. Select Mainnet
2. Go to https://etherscan.io/token/0xdac17f958d2ee523a2206206994597c13d831ec7#writeContract
3. Connect MM
4. Click Deprecate
5. Add an address
6. Check MM

### Version

10.24.0

### Build type

None

### Browser

Chrome

### Operating system

Linux

**Comment by bschorchit** [comment]
Oh, I see a warning in your first screenshot and also got one when going through the repro steps (screenshot). I think there are some other screens where no warning is being shown.

![Screenshot 2023-01-24 at 12 51 51](https://user-images.githubusercontent.com/34968356/214341966-82ac5856-f2e7-4ef9-9b0a-3d49d023b422.png)

**Comment by bschorchit** [comment]
Finding from Dan M.:

> So we still have this warning, and it is still working, but NOT if the new EIP-1559 gas UI setting is set to true
![Screenshot from 2023-01-25 08-36-45](https://user-images.githubusercontent.com/34968356/214560118-83266e8c-b40b-41c1-a101-f174ac5a7d6b.png)

**Comment by danjm** [comment]
I am 95% sure that the root cause of this is the same as the root cause of https://github.com/MetaMask/metamask-extension/issues/17377

**Comment by danjm** [comment]
@VSaric I am assigning you to this as well, because I think that it will be resolved by the same fix that resolves #17377

cc @ElvirCe 

Also, I am pretty sure that the problematic section of code is in `confirm-transaction-base.component.js`:
```
rows={[
            renderSimulationFailureWarning &&
              !this.supportsEIP1559V2 &&
              simulationFailureWarning(),
            !renderSimulationFailureWarning &&
              !isMultiLayerFeeNetwork &&
              renderGasDetailsItem(),
            !renderSimulationFailureWarning && isMultiLayerFeeNetwork && (
              <MultiLayerFeeMessage
                transaction={txData}
                layer2fee={hexMinimumTransactionFee}
                nativeCurrency={nativeCurrency}
              />
            ),
```

```
